store/store.go

```diff
@@ -425,7 +434,30 @@ func (s *store) Prune(resultFunc func(pruned, skipped, pruningHeight uint32)) er
 	return nil
 }
 
-func (s *store) pruneBlock(blockHeight uint32) (bool, error) { //nolint
+func (s *store) PruneOnCommit() error {
```

Comment:
This logic should move to commit block function.

Reply:
But we have `RetentionBlocks` in the store, and `isPruned` can be determined properly in the store only.

Any way to have them in state as well without issues?

Reply:
We can export a function, `IsPruned`, to inform the outside world that the store is in Prune mode. Remember, based on PIP-27, `pruneBlock` is private, so pruning occurs inside the store.